transport/src/main/java/io/netty/channel/SingleThreadIoEventLoop.java

```diff
@@ -157,7 +158,8 @@ protected void run() {
             if (isShuttingDown()) {
                 ioHandler.prepareToDestroy();
             }
-            runAllTasks(maxTasksPerRun);
+            // Now run all tasks for the maximum configured amount of time before trying to run IO again.
+            runAllTasks(maxTaskDurationPerRun);
```

Comment:
this is a best effect (soft limit), not a hard limit too, I'm not sure the property name

Reply:
I am open for better naming.